candlelight.go

```diff
@@ -36,6 +36,9 @@ var (
 // A different provider can be used if a constructor for it is provided in the
 // config.
 func ConfigureTracerProvider(config Config) (trace.TracerProvider, error) {
+	if config.Default {
```

Comment:
this change doesn't line up with https://github.com/xmidt-org/tr1d1um/issues/203#issuecomment-803158742.

Specifically: 
 - > No tracer is provided in configuration
   - if no config is provided no tracing should be used. 
 - > The client explicitly asks for the default tracing behavior
   - if default is set, the default tracer should be used. I don't know what this is maybe `stdout`?

Reply:
I had handled it in 3 ways.

1. if tracing is missing in configuration then the default will be set as true and viper.unmarshal won't run, In this case. NoopTracerProvider will be reurned.
2. if tracing.default is true in configuration then also NoopTracerProvider will be returned.
3. if tracing.provider is defined. and the default is defined as false or skipped the will be returning the tracerProvider based on provider name which was given as input.

![image](https://user-images.githubusercontent.com/12969957/113602370-25cab300-9660-11eb-95d4-539dc7f2f3e4.png)


Now in SetLogger function, we will try to validate if the span is valid or invalid. By invalid I am referring to noopSpan which will be created by points 1 and 2. if the span is invalid then we will not add any trace information in the logger.


![image](https://user-images.githubusercontent.com/12969957/113603110-2152ca00-9661-11eb-9b36-155acadaac0b.png)

Reply:
I don't believe @johnabass intended for the default tracer to be a Noop.

Reply:
We don't want the default tracer to be a noop.  The idea behind a client explicitly opting in to a default is that the client *does* want tracing, but s/he just wants to use whatever tracing is provided out of the box.  This is a distinct case from not supplying any configuration or turning off tracing.

We should definitely keep the concept of a noop tracer, though.  It allows a client to explicitly configure a noop.  When you deal with a large number of config files, it's handy do things like:

```
tracing:
  # example comment: we don't want tracing in a dev environment
  # when you're trying to figure out why no tracing is happening, this is nice
  noop: true
```

and

```
tracing:
  # example comment: this is what we use in XYZ environment
  default: true
```

I'm ok with not supporting `noop` via configuration yet.  We could add that as a feature issue.

Reply:
Hi @johnabass 
we can put the noop as value of `tracing.provider` then we don't need to provide specially noop inside tracing as `tracing.noop`

Reply:
Hi @johnabass, @kcajmagic 

Do i need to handle it in these 3 ways
1. A tracer is provided in the configuration: If the configuration isn't valid, an error should result.
2. No tracer is provided in the configuration: Assuming it as Noop and skipping addition of traceId and spanId in logs.
3. The client explicitly asks for the default tracing behavior: If the default is true then I will be overriding the `tracing.provider` to `stdout` and will set `tracing.skipTraceExport` to `true` which is the default one.

Why `stdout` because in this we don't need any external endpoint to post the trace information.

Reply:
@Sachin4403, I'm working on the first two cases here in case you want to take a look: https://github.com/xmidt-org/tr1d1um/pull/207/files

I'm also making some changes to candlelight so it might be a good idea to put a hold on this PR for now to avoid double work. We can build on top of the changes I'm making.